python-packages/smithy-http/smithy_http/__init__.py

```diff
@@ -172,6 +172,12 @@ def __eq__(self, other: object) -> bool:
     def __iter__(self) -> Iterator[interfaces.Field]:
         yield from self.entries.values()
 
+    def __len__(self) -> int:
+        return len(self.entries)
+
+    def __repr__(self) -> str:
```

Comment:
I think we should add a test for this similar to the one you added for `__len__`